src/algebra/star/module.lean

```diff
@@ -33,3 +34,60 @@ def star_linear_equiv (R : Type*) {A : Type*}
 { to_fun := star,
   map_smul' := star_smul,
   .. star_add_equiv }
+
+variables (R : Type*) (A : Type*)
+  [semiring R] [star_monoid R] [has_trivial_star R]
+  [add_comm_group A] [module R A] [star_add_monoid A] [star_module R A]
+
+/-- The self-adjoint elements of a star module, as a submodule. -/
+def self_adjoint.submodule : submodule R A :=
+{ smul_mem' := self_adjoint.smul_mem,
+  ..self_adjoint A }
+
+/-- The skew-adjoint elements of a star module, as a submodule. -/
+def skew_adjoint.submodule : submodule R A :=
+{ smul_mem' := skew_adjoint.smul_mem,
+  ..skew_adjoint A }
+
+variables {A} [invertible (2 : R)]
+
+/-- The self-adjoint part of an element of a star module, as a linear map. -/
+@[simps] def self_adjoint_part : A →ₗ[R] self_adjoint A :=
+{ to_fun := λ x, ⟨(⅟2 : R) • (x + star x),
+  by simp only [self_adjoint.mem_iff, star_smul, add_comm,
+                  star_add_monoid.star_add, star_inv', star_bit0,
+                  star_one, star_star, star_inv_of (2 : R), star_trivial]⟩,
+  map_add' := λ x y, by { ext, simp [add_add_add_comm] },
+  map_smul' := λ r x, by { ext, simp [←mul_smul,
+          show ⅟ 2 * r = r * ⅟ 2, from commute.inv_of_left (commute.one_left r).bit0_left] } }
+
+/-- The skew-adjoint part of an element of a star module, as a linear map. -/
+@[simps] def skew_adjoint_part : A →ₗ[R] skew_adjoint A :=
+{ to_fun := λ x, ⟨(⅟2 : R) • (x - star x),
+    by simp only [skew_adjoint.mem_iff, star_smul, star_sub, star_star, star_trivial, ←smul_neg,
+                  neg_sub]⟩,
+  map_add' := λ x y, by { ext, simp only [sub_add, ←smul_add, sub_sub_assoc_swap, star_add,
+                                          add_subgroup.coe_mk, add_subgroup.coe_add] },
+  map_smul' := λ r x, by { ext, simp [←mul_smul, ←smul_sub,
+            show r * ⅟ 2 = ⅟ 2 * r, from commute.inv_of_right (commute.one_right r).bit0_right] } }
+
+lemma star_module.self_adjoint_part_add_skew_adjoint_part (x : A) :
+  (self_adjoint_part R x : A) + skew_adjoint_part R x = x :=
+by simp only [smul_sub, self_adjoint_part_apply_coe, smul_add, skew_adjoint_part_apply_coe,
+              add_add_sub_cancel, inv_of_two_smul_add_inv_of_two_smul]
+
+variables (A)
+
+/-- The decomposition of elements of a star module into their self- and skew-adjoint parts,
+as a linear equivalence. -/
+def star_module.decompose_prod_adjoint : A ≃ₗ[R] self_adjoint A × skew_adjoint A :=
+linear_equiv.of_linear
+(linear_map.prod (self_adjoint_part R) (skew_adjoint_part R))
+(linear_map.coprod (self_adjoint.submodule R A).subtype (skew_adjoint.submodule R A).subtype)
+(by ext; simp)
+begin
+  rw [linear_map.coprod_comp_prod],
+  ext,
+  have : (⅟2 : R) • (x + x) = x := by rw [smul_add, inv_of_two_smul_add_inv_of_two_smul],
+  simp [←smul_add, this],
+end
```

Comment:
Is one of these two proofs `self_adjoint_part_add_skew_adjoint_part`?

Reply:
I think simp just reproves it somehow.